pages/accountLists/[accountListId]/contacts/[[...contactId]].page.tsx

```diff
@@ -60,12 +79,42 @@ const ContactsPage: React.FC = () => {
     setContactDetailsOpen(!!id);
   };
 
+  const handleCheckOneContact = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    contactId: string,
+  ): void => {
+    if (!selectedContacts.includes(contactId)) {
+      setSelectedContacts((prevSelected) => [...prevSelected, contactId]);
+    } else {
+      setSelectedContacts((prevSelected) =>
+        prevSelected.filter((id) => id !== contactId),
+      );
+    }
+  };
+
+  const handleCheckAllContacts = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ): void => {
+    setSelectedContacts(
+      event.target.checked
+        ? data?.contacts.nodes.map((contact) => contact.id) ?? []
+        : [],
+    );
+  };
+
+  const isSelectedSomeContacts =
+    selectedContacts.length > 0 &&
+    selectedContacts.length < (data?.contacts.nodes.length ?? 0);
+  const isSelectedAllContacts =
+    selectedContacts.length === data?.contacts.nodes.length;
```

Comment:
Maybe nitpicky but I feel like ```hasSelected...``` would be be better wording than ```isSelected...```

Reply:
That was copied from work David did. We need to make mass selection's select all support the current contact filters since we won't always have all contacts and therefore a complete list of ids loaded. It's future work but I approved this initial implementation for now. I'll leave this alone for now.